spatial_compare/utils.py

```diff
@@ -48,16 +49,109 @@ def grouped_obs_mean(adata, group_key, layer=None, gene_symbols=None):
     return out
 
 
+def spatial_detection_score_kde(query: pd.DataFrame, grid_out: int = 100):
+    cell_x = query["x_centroid"]
+    cell_y = query["y_centroid"]
+    cell_coords = np.vstack([cell_x.values, cell_y.values])
+
+    xmin, xmax = cell_x.min(), cell_x.max()
+    ymin, ymax = cell_y.min(), cell_y.max()
+    extent = [xmin, xmax, ymin, ymax]
+
+    if grid_out == 0:
+        positions = cell_coords
+    else:
+        X, Y = np.mgrid[
+            xmin : xmax : complex(0, grid_out), ymin : ymax : complex(0, grid_out)
+        ]
+        positions = np.vstack([X.ravel(), Y.ravel()])
+
+    scores = []
+    for column in [
+        "detection_relative_z_score",
+        "detection_difference",
+        "log_10_detection_ratio",
+    ]:
+        weights = query[column].values
+        weights[np.isnan(weights)] = 0
+
+        wext = [weights.min(), weights.max()]
+
+        weights = weights + abs(weights.min())
+
+        kde_weighted = sp.stats.gaussian_kde(cell_coords, weights=weights)
+        kde_unweighted = sp.stats.gaussian_kde(
+            cell_coords,
+        )
+
+        if column == "detection_relative_z_score":
+            estimator = (
+                lambda positions: kde_weighted(positions).T
+                / kde_unweighted(positions).T
+            )
+
+        Z = kde_weighted(positions).T / kde_unweighted(positions).T
+        Z = (Z - Z.min()) / (Z.max() - Z.min()) * (wext[1] - wext[0]) + wext[0]
```

Comment:
@berl re: plotting artifact... I added a linear rescaling here to get it back into approximately the same range as the binned outputs. Maybe not exactly the right treatment, but I would suspect it doesn't match 1:1 with the binned version

Reply:
got it, thanks!  I think the z-score plot values should reflect the true relationship to the reference distribution and only tweak the plot ranges for visualization if needed. I can revisit this in #36